app/models/ssh_key.rb

```diff
@@ -6,7 +6,7 @@ class SshKey < ApplicationRecord
   include Parameterizable::ByIdName
 
   belongs_to :user
-  before_validation :generate_fingerprint
+  before_validation :generate_fingerprint, on: :create
```

Comment:
This should still run on update as well, in case a user changes their key we need to update the fingerprint as well.